app/main/views/send.py

```diff
@@ -606,17 +617,51 @@ def check_messages(service_id, template_id, upload_id, row_index=2):
         metadata_kwargs["sender_id"] = session["sender_id"]
 
     set_metadata_on_csv_upload(service_id, upload_id, **metadata_kwargs)
-
     return render_template("views/check/ok.html", **data)
 
 
+@main.route(
+    "/services/<uuid:service_id>/<uuid:template_id>/check/<uuid:upload_id>/preview",
+    methods=["GET"],
+)
+@main.route(
+    "/services/<uuid:service_id>/<uuid:template_id>/check/<uuid:upload_id>/preview/row-<int:row_index>",
+    methods=["GET"],
+)
+@user_has_permissions("send_messages", restrict_admin_usage=True)
+def preview_job(service_id, template_id, upload_id, row_index=2):
+    session["scheduled_for"] = request.args.get("scheduled_for", "")
+    data = _check_messages(service_id, template_id, upload_id, row_index)
+    data["allowed_file_extensions"] = Spreadsheet.ALLOWED_FILE_EXTENSIONS
+    if (
```

Comment:
I added these errors between lines 635-651 to catch any errors. But the function check_messages already checks the error. Would you think it's necessary to have it again? I included the errors because it wouldn't hurt to have another check, but also seems redundant. Feedback welcomed